modules/azure-oracle-adbs/main.tf

```diff
@@ -0,0 +1,77 @@
+terraform {
+  required_providers {
+    azapi = {
+      source = "Azure/azapi"
+    }
+    azurerm = {
+      source = "hashicorp/azurerm"
+    }
+  }
+}
+
+# uncomment for local module run
+# provider "azurerm" {
+#   features {}
+# }
+
+# provider "azapi" {}
+
+
+data "azurerm_virtual_network" "virtual_network" {
+  name                = var.nw_vnet_name
+  resource_group_name = var.nw_resource_group
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/subnet
+data "azurerm_subnet" "subnet" {
+  name                 = var.nw_delegated_subnet_name
+  virtual_network_name = var.nw_vnet_name
+  resource_group_name  = var.nw_resource_group
+}
+
+
+## DB resources 
+data "azurerm_resource_group" "resource_group" {
+  name = var.db_resource_group
+}
+
+
+resource "azapi_resource" "autonomous_db" {
+  type                      = "Oracle.Database/autonomousDatabases@2023-09-01"
+  parent_id                 = data.azurerm_resource_group.resource_group.id
+  name                      = var.db_name
+  schema_validation_enabled = false
+
+  timeouts {
+    create = "3h"
+    update = "2h"
+    delete = "1h"
+  }
+
+  body = jsonencode({
+    "location" : var.location,
+    "properties" : {
+      "displayName" : var.db_name,
+      "computeCount" : var.db_ecpu_count,
+      "dataStorageSizeInGbs" : var.db_storage_in_gb,
```

Comment:
I am still confused about this field v.s. `dataStorageSizeInGbs `. Do both of them work for both ADW and ATP?

Reply:
I tested DW with both  at a time  and it worked fine in either case. Since we are not DB experts we will stick to quick start terraform templating & let db expert configured all properties and its values based on need.